### PR TITLE
Feature/recaptcha

### PR DESCRIPTION
**Pull request by TanyaDovzhenko**

### CODE REVIEW

File: src/components/auth/recover-password-form.tsx [hunk 1]
```diff
@@ -4,6 +4,9 @@ import { Box, Button, TextField, Typography } from '@mui/material';
 import inputStyles from '../../styles/input-styles';
 import { Formik, Form, Field } from 'formik';
 import userService from '../../api/user.service';
+import { useGoogleReCaptcha } from "react-google-recaptcha-v3";
```
Comment (RamboUA) [inline src/components/auth/recover-password-form.tsx:7]: Better to import libs on the very top

**Review by RamboUA — APPROVED** [review]

File: src/utils/recaptcha.ts [hunk 2]
```diff
@@ -0,0 +1,3 @@
+export async function recaptchaVerify(executeRecaptcha: any) {
+    if (executeRecaptcha) return await executeRecaptcha('action')
```
Comment (BillyAleks) [inline src/utils/recaptcha.ts:2]: for what do we need await here?

**Review by BillyAleks — CHANGES_REQUESTED** [review]

**Review by BillyAleks — COMMENTED** [review]
Thanks, please check the comments

File: src/components/auth/sign-in-form.tsx [hunk 3]
```diff
@@ -30,55 +32,56 @@ const validationSchema = yup.object({
...
-          <Typography sx={{ textAlign: 'center', fontSize: '2rem' }}>
-            Sign In
-          </Typography>
+      <Box sx={formStyles}>
+        <Typography sx={{ textAlign: 'center', fontSize: '2rem' }}>
+          Sign In
+        </Typography>
 
-          {isError && <Alert severity='error'>Wrong email or password</Alert>}
+        {isError ? (
```
Comment (RamboUA) [inline src/components/auth/sign-in-form.tsx:70]:
```suggestion
        {isError && <Alert severity='error'>Wrong email or password</Alert>}
```

File: src/components/auth/sign-in-form.tsx [hunk 4]
```diff
@@ -30,55 +32,56 @@ const validationSchema = yup.object({
...
+        <Formik
+          initialValues={initialValues}
+          validationSchema={validationSchema}
+          onSubmit={async (values, formikHelpers) => {
+            setDisableSend(true)
+            await signIn(values).finally(() => setDisableSend(false))
+            formikHelpers.resetForm();
+          }}
+        >
+          {({ errors, touched, isValid, dirty }) => {
```
Comment (RamboUA) [inline src/components/auth/sign-in-form.tsx:83]:
```suggestion
          {({ errors, touched, isValid, dirty }) => (
              <Form
                style={{
                  display: 'flex',
                  ...
```

File: src/pages/auth/reset-password.tsx [hunk 5]
```diff
@@ -1,25 +1,30 @@
 import * as yup from 'yup'
 import { useFormik } from 'formik'
 import { useState } from 'react'
-import userService from '../api/user.service'
+import userService from '../../api/user.service'
 import { Box, Button, Input, Typography } from '@mui/material'
 import { useNavigate, useSearchParams } from 'react-router-dom'
+import { useGoogleReCaptcha } from 'react-google-recaptcha-v3'
```
Comment (RamboUA) [inline src/pages/auth/reset-password.tsx:7]: Maybe we could add a rule of imports. It looks better when libs are imported on a top level

File: src/pages/auth/reset-password.tsx [hunk 6]
```diff
@@ -1,25 +1,30 @@
...
   const [success, setSuccess] = useState(false)
   const [error, setError] = useState(false)
   const [resetToken, setResetToken] = useSearchParams()
+  const { executeRecaptcha } = useGoogleReCaptcha()
 
   const navigate = useNavigate()
 
   const formik = useFormik({
     initialValues: { newPassword: '', newPasswordConfirm: '' },
     onSubmit: async (values) => {
```
Comment (RamboUA) [inline src/pages/auth/reset-password.tsx:21]: Better to do it as a separate function

**Review by RamboUA — CHANGES_REQUESTED** [review]
Thanks, please check the comment

**Pull request merged**